src/list-data/request-action-mapping.ts

```diff
@@ -61,32 +61,31 @@ export const BcMapping: RequestActionMappingI = {
 
 export const XproMapping: RequestActionMappingI = {
   AML: [
-    EntityTypes.XCR,
-    EntityTypes.XCP
+    EntityTypes.XCP, // maps to Corp Type "XCP"
+    EntityTypes.XCR // maps to Corp Type "XCR"
   ],
   ASSUMED: [
-    EntityTypes.XCR,
-    EntityTypes.RLC
+    EntityTypes.XCR, // maps to Corp Type "XCR"
+    EntityTypes.RLC // maps to Corp Type "LLC"
   ],
   REN: [
-    EntityTypes.XCR,
-    EntityTypes.XCP,
-    EntityTypes.RLC
+    EntityTypes.RLC, // maps to Corp Type "LLC"
+    EntityTypes.XCP, // maps to Corp Type "XCP"
+    EntityTypes.XCR // maps to Corp Type "XCR"
   ],
   REH: [
-    EntityTypes.A,
-    EntityTypes.LLC,
-    EntityTypes.XCP,
-    EntityTypes.XSO
+    EntityTypes.RLC, // maps to Corp Type "LLC"
+    EntityTypes.XCP, // maps to Corp Type "XCP"
+    EntityTypes.XSO, // maps to Corp Type "XS"
+    EntityTypes.XUL // maps to Corp Type "A"
   ],
   CHG: [
-    EntityTypes.A,
-    EntityTypes.LLC,
-    EntityTypes.XCP,
-    EntityTypes.XL,
-    EntityTypes.XLL,
-    EntityTypes.XLP,
-    EntityTypes.XP
+    EntityTypes.RLC, // maps to Corp Type "LLC"
+    EntityTypes.XCP, // maps to Corp Type "XCP"
+    EntityTypes.XLL, // maps to Corp Type "XL"
+    EntityTypes.XLP, // maps to Corp Type "XP"
+    EntityTypes.XSO, // maps to Corp Type "XS"
+    EntityTypes.XUL // maps to Corp Type "A"
```

Comment:
The root case of this bug is that these lists use Namerequest-specific codes, but we check the legal type (corp type) from the business search result, which is LEAR codes.

So we need to keep in mind this mapping between the codes 😒

Reply:
I really really wish we didn't have this mismatch between `EntityTypes` and `CorpTypeCd` 😞

Reply:
I know. This will keep biting us.

I just created this ticket:
https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/18241

Reply:
I agree as well, would be really nice to have the codes consistent.